srcCpp/RTIRawTransportImpl.cxx

```diff
@@ -677,6 +649,14 @@ class RTIRawTransportSubscriber : public IMessagingReader
             _message.size = _data.bin_data.length();
             _message.data = (char *)_data.bin_data.get_contiguous_bufferI();
 
+            /*
```

Comment:
If we are using batching, the buffer will have N samples stored one after other.
I need to take one sample each time. The problem is that I need to know how much the sample occupied on that buffer to set an offset and deserialize again from that point on the buffer.
Before I was using the RTICdrStream that was doing this by his own.

Now I could use serialize_data_to_cdr_buffer() with the buffer set it to null In order to retrieve the max size serialized sample, but as I read on the documentation, this size could be bigger than the actual size of the serialized sample. https://community.rti.com/static/documentation/connext-dds/6.0.0/doc/api/connext_dds/api_cpp/structFooTypeSupport.html#a7b6d1dec0082d30804687c145f41c9c7

So, in order to know exactly the size of the serialized sample, I calculate it on my own because the serialized sample size will be always _message.size + perftest_cpp::OVERHEAD_BYTES + RTI_CDR_ENCAPSULATION_HEADER_SIZE;

Reply:
Now thinking more about this, I check that the size for RTI_XCDR_ENCAPSULATION_HEADER_SIZE and the one for 
CDR is the same (4 bytes) but could this change on the future?. 
For now, this works but I don't want to have it as a temporary solution.

Reply:
@alfonso-soto Thanks for explaining. RTI_XCDR_ENCAPSULATION_HEADER_SIZE and the one for CDR will always be the same.